core/src/main/java/greencity/service/impl/CloudStorageServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package greencity.service.impl;
+
+import com.google.cloud.WriteChannel;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import greencity.service.FileService;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service
+public class CloudStorageServiceImpl implements FileService {
+    final String projectId;
+    final StorageOptions.Builder optionsBuilder;
+    private final String bucketName;
+    private final Storage storage;
+
+    /**
+     * Constructor with parameters.
+     */
+    public CloudStorageServiceImpl(@Value("${PROJECT_ID}") final String projectId,
+                                   @Value("${BUCKET_NAME}") final String bucketName) {
+        this.projectId = projectId;
+        this.bucketName = bucketName;
+        this.optionsBuilder = StorageOptions.newBuilder();
+        if (bucketName == null || bucketName.isEmpty()) {
+            throw new IllegalArgumentException("Bucket name is null");
+        }
+        if (projectId != null && !projectId.isEmpty()) {
+            optionsBuilder.setProjectId(projectId);
+        }
+        this.storage = optionsBuilder.build().getService();
+    }
+
+
+    @Override
+    public URL upload(final MultipartFile multipartFile) throws IOException {
+        try {
+            final String contentType = multipartFile.getContentType();
+            final String blob = UUID.randomUUID().toString();
+            final BlobInfo blobInfo = BlobInfo.newBuilder(bucketName, blob).setContentType(contentType).build();
+
+            return write(storage, blobInfo, multipartFile);
+        } catch (IOException ex) {
+            throw ex;
+        }
```

Comment:
Do you really need to handle IOException ?

Reply:
And you don't need to use final for every variable. Please, use it only when you explicitly don't want it to be changed

Reply:
The reason I use final everywhere in my code:
1. This improves readability of the code;
2. Code is better structured;
3. It gives hints JVM to produced optimized CPU instructions.

Reply:
https://stackoverflow.com/questions/316352/why-would-one-mark-local-variables-and-method-parameters-as-final-in-java